extension/src/editors/form-editor/webview-communication.ts

```diff
@@ -24,3 +24,16 @@ export const setupCommunication = (messenger: Messenger, webviewPanel: vscode.We
   );
   webviewPanel.onDidDispose(() => toDispose.dispose());
 };
+
+export class FormEditorWebSocketForwarder extends WebSocketForwarder {
+  protected override handleClientMessage(message: unknown) {
+    if (this.isSaveData(message)) {
+      message.method = 'saveDataAndBuild';
+    }
+    super.handleClientMessage(message);
+  }
+
+  isSaveData = (obj: unknown): obj is { method: string } => {
+    return typeof obj === 'object' && obj !== null && 'method' in obj && obj.method === 'saveData';
+  };
+}
```

Comment:
with this you may can even implement the dirtyflag mode.
1. By catch the saveData message and hold the payload 
2. Set the dirty state to true
3. On real editor save (by the user ctrl + s) -> send saveDataAndBuild with the last saveData payload